src/evict/evict_page.c

```diff
@@ -115,6 +117,18 @@ __wt_evict(WT_SESSION_IMPL *session, WT_REF *ref, uint8_t previous_state, uint32
      * generation (which must be as low as the current generation), untouched.
      */
     if (__wt_session_gen(session, WT_GEN_EVICT) == 0) {
+        /*
```

Comment:
I'd recommend pulling the cursor open/close calls outside the test for the eviction generation. It's not wrong to do it here, but it's simpler to always do the call pair rather than micro-optimize based on whether we've already set the generation.

I won't argue if you feel strongly it should remain here.